R/predict_setting_contacts.R

```diff
@@ -36,10 +63,30 @@ predict_setting_contacts <- function(population, contact_model, age_breaks) {
     .f = predictions_to_matrix,
     .options = furrr::furrr_options(seed = TRUE)
   )
-
+  
   combination <- Reduce("+", setting_matrices)
   setting_matrices$all <- combination
   
-  setting_matrices
+  # if we haven't set anything for the per capita household size, return this
+  # adjusted matrix
+  if (is.null(per_capita_household_size)) {
+    return(setting_matrices)
+  
+  # otherwise we want to adjust the household contact matrix (which also 
+    # updates the "all" setting matrix).
+  } else if (!is.null(per_capita_household_size)){
+    setting_matrices <- adjust_household_contact_matrix(
+      setting_matrices = setting_matrices,
+      # how do we choose this household size?
+      household_size = per_capita_household_size,
+      population = population,
+      # extra arguments to adjust_household_contact_matrix
+      ...
+      )
+    return(
+      setting_matrices
+    )
```

Comment:
@goldingn does this match what you were thinking? At the moment `household_size` here is just a single number, I'm not sure if you wanted this to be able to be a distribution/vector of numbers?

Reply:
`get_per_capita_household_size()` looks up the distribution and turns it into a single number, so this is right!

It would be helpful to have details either here, in the docs for  `get_per_capita_household_size()`, or both, to explain what the per-capita household size is and how it differs to the mean household size (it's the household size averaged over people in the population, not over households, so that larger households get upweighted), and how it is calculated (by taking a distribution of the number of households of each size in a population, multiplying the size by the household by the household count to get the number of people with that size of household, and computing the population-weighted average of household sizes), and why we do that (because it is a more accurate reflection of the average number of household members a person in the population can have contact with)

Reply:
OK, thanks, Nick! I'm adding this in now, basically just paraphrasing what you said here - thanks for that detail!

Reply:
Done.